functional-tests/tests/tests.js

```diff
@@ -25,68 +23,60 @@ module.exports = {
 		browser.end();
 	},
 	'Check Tag in search box for URL with tag filter': function (browser) {
-		browser.url("http://bagcupid.com/app/tags/3")
-		browser.pause(7000);
-		browser.assert.visible(".select2-container--default .select2-selection--multiple .select2-selection__choice");
+	    browser.url("http://bagcupid.com/app/tags/3");
+	    browser.waitForElementVisible(".select2-container--default .select2-selection--multiple .select2-selection__choice", 7000);
 		browser.end();
 	},
 	'Check for "About US" screen': function (browser) {
 		browser.url("https://bagcupid.com/app/aboutus");
-		browser.pause(7000);
-		browser.assert.visible(".about-section");
+		browser.waitForElementVisible(".about-section", 7000);
 		browser.end();
 	},
 	'Banner should be hidden for URL with filter': function (browser) {
 		browser.url("http://bagcupid.com/app/tags/3");
-		browser.pause(6000);
-		browser.expect.element('.banner').to.not.be.visible;
+		browser.waitForElementNotVisible(".banner", 7000);
 		browser.end();
 	},
 	'Check for Bag images slider': function (browser) {
 		browser.url(process.env.BAG_CUPID_URL);
-		browser.pause(7000);
+		browser.waitForElementVisible('.product-list .product-card:first-child', 7000);
 		browser.click('.product-list .product-card:first-child .owl-next');
-		browser.pause(2000);
-		browser.assert.visible(".product-list .product-card:first-child .owl-wrapper > .owl-item:nth-child(2) img");
+		browser.waitForElementVisible(".product-list .product-card:first-child .owl-wrapper > .owl-item:nth-child(2) img", 2000);
 		browser.click('.product-list .product-card:first-child .owl-next');
-		browser.pause(2000);
-		browser.assert.visible(".product-list .product-card:first-child .owl-wrapper > .owl-item:nth-child(3) img");
+		browser.waitForElementVisible(".product-list .product-card:first-child .owl-wrapper > .owl-item:nth-child(3) img", 2000);
 		browser.end();
 	},
-	'Banner should be hidden for URL with filter': function (browser) {
-		browser.url("http://bagcupid.com/app/tags/3");
-		browser.pause(7000);
+	'Side filter should open': function (browser) {
+	    browser.url(process.env.BAG_CUPID_URL);
+		browser.waitForElementVisible("#side-filter-trigger",7000);
 		browser.click("#side-filter-trigger");
 		browser.pause(2000);
```

Comment:
Here I added a test on "side bar filter" to ensure it opens when clicking on Filter button in header. I have to use `pause` here because I can't use any other alternative here. Nothing else happens on UI apart from the side bar's `style` get changed